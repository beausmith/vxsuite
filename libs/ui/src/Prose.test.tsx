import React from 'react';
import { render } from '@testing-library/react';

import { Prose } from './Prose';

const proseContent = (
  <React.Fragment>
    <h1>Heading 1</h1>
    <h2>Heading 2</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h1>Heading 1</h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h2>Heading 2</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h3>Heading 3</h3>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
  </React.Fragment>
);
describe('renders Prose', () => {
  test('with defaults', async () => {
    const { container } = render(<Prose>{proseContent}</Prose>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('with with non-default options', async () => {
    const { container } = render(
      <Prose compact textCenter maxWidth={false}>
        {proseContent}
      </Prose>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('with right-aligned text', async () => {
    const { container } = render(<Prose textRight>{proseContent}</Prose>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('with theme', async () => {
    const { container } = render(
      <Prose
        theme={{
          fontSize: '10px',
          color: '#666666',
        }}
      >
        {proseContent}
      </Prose>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
