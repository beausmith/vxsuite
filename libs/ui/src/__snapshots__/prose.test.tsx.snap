// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Prose with defaults 1`] = `
<div
  class="sc-bdvvaa bnaJnM"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;

exports[`renders Prose with right-aligned text 1`] = `
<div
  class="sc-bdvvaa hNfOpU"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;

exports[`renders Prose with theme 1`] = `
<div
  class="sc-bdvvaa cLgjBe"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;

exports[`renders Prose with with non-default options 1`] = `
<div
  class="sc-bdvvaa bAMuWr"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;
